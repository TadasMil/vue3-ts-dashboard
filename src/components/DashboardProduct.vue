<script setup lang="ts" name="DashboardProduct">
const { title, description, rating, price } = defineProps<{
  title: string
  description: string
  rating: number
  price: number
}>()

defineOptions({
  inheritAttrs: false
})
</script>

<template>
  <div class="product-box gray-box">
    <div class="product-box__title-description">
      <span class="title">{{ title }}</span>
      <span class="description">{{ description }}</span>
    </div>

    <span>{{ rating }}</span>
    <span>{{ price }}$</span>
  </div>
</template>

<style scoped>
.product-box {
  gap: 16px;
  display: grid;
  border-radius: 6px;
  align-items: center;
  grid-template-columns: 1fr auto auto;
  justify-content: space-between;
}

.product-box__title-description {
  display: flex;
  gap: 5px 0;
  flex-direction: column;
  flex: 1;
}

.title {
  color: var(--secondary-title-color);
}

.description {
  font-size: 12px;
  color: var(--color-secondary);
}
</style>
