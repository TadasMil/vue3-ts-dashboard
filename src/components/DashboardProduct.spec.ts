import { render } from '@testing-library/vue'
import { describe, expect, it } from 'vitest'

import DashboardProduct from './DashboardProduct.vue'

describe('DashboardProduct.vue', () => {
  it('should render title', () => {
    const { html } = render(DashboardProduct, {
      props: {
        title: 'Title',
        description: 'Description',
        rating: 4.2,
        price: 200
      }
    })

    expect(html()).toMatchSnapshot()
  })
})
