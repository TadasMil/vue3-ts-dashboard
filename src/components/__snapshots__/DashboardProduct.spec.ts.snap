// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashboardProduct.vue > should render title 1`] = `
"<div data-v-f5ca514c=\\"\\" class=\\"product-box gray-box\\">
  <div data-v-f5ca514c=\\"\\" class=\\"product-box__title-description\\"><span data-v-f5ca514c=\\"\\" class=\\"title\\">Title</span><span data-v-f5ca514c=\\"\\" class=\\"description\\">Description</span></div><span data-v-f5ca514c=\\"\\">4.2</span><span data-v-f5ca514c=\\"\\">200$</span>
</div>"
`;
